src/cryptography/hazmat/bindings/openssl/ssl.py

```diff
@@ -602,6 +629,15 @@
 long (*SSL_get_server_tmp_key)(SSL *, EVP_PKEY **) = NULL;
 #endif
 
+/* OpenSSL 1.0.2+ cert validation callback */
+#if OPENSSL_VERSION_NUMBER < 0x10002000L
```

Comment:
This will fail on libressl (which stupidly defines openssl version as 0x20000000L). You'll also need to add `&& !defined(LIBRESSL_VERSION_NUMBER)`